Extension/src/LanguageServer/extension.ts

```diff
@@ -934,6 +1048,67 @@ function onTakeSurvey(): void {
     vscode.commands.executeCommand('vscode.open', uri);
 }
 
+function onVcpkgOnlineHelpSuggested(dummy?: any): void {
+    telemetry.logLanguageServerEvent('vcpkgAction', { 'source': dummy ? 'CodeAction' : 'CommandPalette', 'action': 'vcpkgOnlineHelpSuggested' });
+    const uri: vscode.Uri = vscode.Uri.parse(`https://aka.ms/vcpkg`);
+    vscode.commands.executeCommand('vscode.open', uri);
+}
+
+async function onVcpkgClipboardInstallSuggested(ports?: string[]): Promise<void> {
+    onActivationEvent();
+    let source: string;
+    if (ports && ports.length) {
+        source = 'CodeAction';
+    } else {
+        source = 'CommandPalette';
+        // Glob up all existing diagnostics for missing includes and look them up in the vcpkg database
+        const missingIncludeLocations: [vscode.TextDocument, number[]][] = [];
+        vscode.languages.getDiagnostics().forEach(uriAndDiagnostics => {
+            // Extract textDocument
+            const textDocument: vscode.TextDocument = vscode.workspace.textDocuments.find(doc => doc.uri.fsPath === uriAndDiagnostics[0].fsPath);
+            if (!textDocument) {
+                return;
+            }
+
+            // Extract lines numbers for missing include diagnostics
+            let lines: number[] = uriAndDiagnostics[1].filter(isMissingIncludeDiagnostic).map<number>(d => { return d.range.start.line; });
+            if (!lines.length) {
+                return;
+            }
+
+            // Filter duplicate lines
+            lines = lines.filter((line: number, index: number) => {
+                const foundIndex: number = lines.indexOf(line);
+                return foundIndex === index;
+            });
+
+            missingIncludeLocations.push([textDocument, lines]);
+        });
+        if (!missingIncludeLocations.length) {
+            return;
+        }
+        
+        // Queue look ups in the vcpkg database for missing ports; filter out duplicate results
+        let portsPromises: Promise<string[]>[] = [];
+        missingIncludeLocations.forEach(docAndLineNumbers => {
+            docAndLineNumbers[1].forEach(async line => {
+                portsPromises.push(lookupIncludeInVcpkg(docAndLineNumbers[0], line));
+            });
+        });
+        ports = [].concat(...(await Promise.all(portsPromises)));
+        if (!ports.length) {
+            return;
+        }
+        ports = ports.filter((port: string, index: number) => { return ports.indexOf(port) === index; });
+    }
+
+    let installCommand: string = 'vcpkg install';
+    ports.forEach(port => installCommand += ` ${port}`);
+    telemetry.logLanguageServerEvent('vcpkgAction', { 'source': source, 'action': 'vcpkgClipboardInstallSuggested', 'ports': ports.toString() });
+
+    return vscode.env.clipboard.writeText(installCommand);
```

Comment:
`onVcpkgClipboardInstallSuggested` returns Thenable<void>.  The right thing to do is `await` the `writeText` function call if you want this function to return a promise that fires after `writeText` completes.